compiler/src/iree/compiler/Codegen/Common/FastMathPatterns.cpp

```diff
@@ -19,109 +21,116 @@ namespace {
 struct FastErfPattern : public OpRewritePattern<math::ErfOp> {
   using OpRewritePattern::OpRewritePattern;
 
+  // Helper to evaluate a polynomial using FMA chains for both scalar and vector
+  // types.
+  template <typename ValueT>
+  static ValueT evalErfPolynomial(ValueT ax, ValueT t, ArrayRef<ValueT> coeffs,
+                                  PatternRewriter &rewriter, Location loc) {
+    ValueT acc = coeffs[0];
+    for (size_t i = 1; i < coeffs.size(); ++i) {
+      acc = rewriter.create<math::FmaOp>(loc, acc, t, coeffs[i]);
+    }
+    return rewriter.create<math::FmaOp>(loc, ax, acc, ax);
+  }
+
   LogicalResult matchAndRewrite(math::ErfOp op,
                                 PatternRewriter &rewriter) const override {
     Location loc = op.getLoc();
     Value input = op.getOperand();
     Type resultType = op.getType();
 
-    // Erf only supports f32.
-    if (!resultType.isF32()) {
-      return rewriter.notifyMatchFailure(op, "Result only supports f32");
+    // Only support f32 and vector<f32> types.
+    if (!(resultType.isF32() ||
+          (llvm::isa<VectorType>(resultType) &&
+           llvm::cast<VectorType>(resultType).getElementType().isF32()))) {
+      return rewriter.notifyMatchFailure(
+          op, "Result only supports f32 or vector<f32>");
     }
```

Comment:
declare `resVecType` variable. please also use it in the lambda.

The comment and error message is misleading. It also supports other vector types that have `f32` element type, but not only `vector<f32>`.